scanner/src/main/kotlin/Scanner.kt

```diff
@@ -141,9 +142,16 @@ abstract class Scanner(val scannerName: String, protected val config: ScannerCon
         // Add scan results from de-duplicated project packages to result.
         consolidatedProjectPackageMap.forEach { (referencePackage, deduplicatedPackages) ->
             resultContainers.find { it.id == referencePackage.id }?.let { resultContainer ->
-                deduplicatedPackages.forEach {
-                    resultContainers += resultContainer.copy(id = it.id)
+                deduplicatedPackages.forEach { dedup ->
+                    analyzerResult.projects.find { it.id == dedup.id }?.let { project ->
```

Comment:
please turn '?' into  '!!' instead of throwing explicitly

Reply:
I prefer throwing explicitly, because it allows setting a custom exception message instead of throwing a generic null pointer exception.